backends/ebpf/codeGen.cpp

```diff
@@ -49,19 +51,13 @@ bool CodeGenInspector::preorder(const IR::Declaration_Variable* decl) {
 
 bool CodeGenInspector::preorder(const IR::Operation_Binary* b) {
     widthCheck(b);
-    int prec = expressionPrecedence;
-    bool useParens = prec > b->getPrecedence();
-    if (useParens)
-        builder->append("(");
+    builder->append("(");
```

Comment:
Perhaps you can review the code and remove all uses of expressionPrecedence and remove the field too.

Reply:
If I remove all uses of expressionPrecedence, then all uses of its comparisons with object->getPrecedence() will be removed as well.  Can't have that or more gcc compiler failures crop up.  It's tricky.

Reply:
With the code changes of this PR to cogeGen.cpp, another test breaks.  It's as follows:

The P4 code below

`if (meta->_acl_outer21 == 1)`

is emitted as 

`if ((meta->_acl_outer21 == 1))`

With double parenthesis, gcc will cause an error.

In a subsequent checkin, I will add the line above and another line that can break gcc compiling with wrong C code generation.  The checkin will also add a fix that fixes all three issues - the `&` line of code has appropriate parenthesis, and so will the two lines of if-statements.